### PR TITLE
Feature/necessary typing

### PR DESCRIPTION
**Pull request by Hadatko**

### CODE REVIEW

File: dissect/cstruct/__init__.py [hunk 1]
```diff
@@ -11,8 +11,15 @@
...
     StructureType,
+    StructureFieldType,
     EnumType,
     ArrayType,
+    PackedType,
+    CharType,
+    WcharType,
+    UnionType,
+    VoidType,
+    typeAll,
```
Comment (qcabrol) [inline dissect/cstruct/__init__.py:22]: typeAll ?

**Comment by qcabrol** [comment]
```
 ✘ quentin@quentin-M17xR4  ~/aeler/dissect.cstruct/tests   devel  pytest test_basic.py 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.9.5, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/quentin/aeler/dissect.cstruct
collected 34 items                                                                                                                                                                                                

test_basic.py ..................................                                                                                                                                                            [100%]

=============================================================================================== 34 passed in 0.60s ================================================================================================
```

I suggest you attach a pipeline if possible to the repo to run it on PRs

**Comment by qcabrol** [comment]
image
This broke the documentation generation apparently

**Review by qcabrol — CHANGES_REQUESTED** [review]
image

Broke the documentation generation. Issue might be on our MsgParser.py side but that somehow breaks retro-compatibilty. One side needs fixing.

Actually this was broken by the previous PR, not this one.

**Comment by qcabrol** [comment]
Regression detected on commit 28f10e0e4bc2ff39c56f9cc96c1f7d5e0668967e

image

File: dissect/cstruct/__init__.py (hunk 1 above)
Comment (Hadatko) [reply dissect/cstruct/__init__.py:22]: typeAll => i created few predefined types to simpliefied typing. This one i exposed.

Comment (Hadatko) [reply dissect/cstruct/__init__.py:22]: image

**Review by qcabrol — APPROVED** [review]

**Comment by qcabrol** [comment]
Solved with changes on the use case code.

**Pull request merged**